src/components/Flow/DecentralizedFlow.tsx

```diff
@@ -1,44 +1,47 @@
 import { Phase } from 'models/FlowPhase'
-import { STATES } from 'models/SigningStates'
 import { errorList } from 'models/ErrorType'
 import { useSnapshot } from 'valtio'
 import AppStore from 'stores/AppStore'
 import JobStore from 'stores/JobStore'
-import Option from 'components/StatusesList/Option'
-import SigningStates from 'models/SigningStates'
+import OptionStatus from 'components/StatusesList/OptionStatus'
+import SigningStates, { States, generatingFlow } from 'models/SigningStates'
 import StatusesList from 'components/StatusesList'
 
 export default function () {
-  const { flowState, error, commitment, proof, phase } = useSnapshot(AppStore)
+  const { flowState, error } = useSnapshot(AppStore)
   const { jobId } = useSnapshot(JobStore)
   const { subTitle } = SigningStates[flowState]
 
-  const statusDescription = error ? errorList[error] : subTitle
+  const serverCaption = jobId
+    ? ' Feel free to leave and come back—we’ll still be here.'
+    : ''
+  const description = subTitle + serverCaption
+  const statusDescription = error ? errorList[error] : description
   const isError = !!error
 
+  const hasCompleteState = (state: States) => {
+    switch (state) {
+      case States.CHECK_COMMITMENT:
+        return !!AppStore.commitment || !!JobStore.jobId
```

Comment:
Can we include `isCompleted` into `SigningStates` file, so we don't need this `switch`?

Reply:
Like this:
```TS
  [States.CHECK_COMMITMENT]: {
    title: 'Checking for existing commitment...',
    completedText: 'Commitment generated',
    subTitle: 'Hang tight!',
    isCompleted: !!AppStore.commitment || !!JobStore.jobId,
  },
```